app/src/main/java/org/eyeseetea/malariacare/data/database/iomodules/dhis/exporter/PushController.java

```diff
@@ -22,167 +22,109 @@
 import android.util.Log;
 
-import com.raizlabs.android.dbflow.sql.language.Delete;
-
-import org.eyeseetea.malariacare.R;
+import org.eyeseetea.malariacare.data.IDataSourceCallback;
 import org.eyeseetea.malariacare.data.database.iomodules.dhis.importer.models.EventExtended;
 import org.eyeseetea.malariacare.data.database.model.Survey;
-import org.eyeseetea.malariacare.data.database.utils.PopulateDB;
-import org.eyeseetea.malariacare.data.remote.PullDhisSDKDataSource;
-import org.eyeseetea.malariacare.receivers.AlarmPushReceiver;
-import org.eyeseetea.malariacare.data.remote.SdkController;
-import org.eyeseetea.malariacare.data.remote.SdkPushController;
+import org.eyeseetea.malariacare.data.database.utils.PreferencesState;
+import org.eyeseetea.malariacare.data.remote.PushDhisSDKDataSource;
+import org.eyeseetea.malariacare.domain.boundary.IPushController;
+import org.eyeseetea.malariacare.domain.exception.ConversionException;
+import org.eyeseetea.malariacare.domain.exception.SurveysToPushNotFoundException;
 import org.eyeseetea.malariacare.utils.Constants;
-import org.hisp.dhis.client.sdk.android.api.persistence.flow.EventFlow;
-import org.hisp.dhis.client.sdk.android.api.persistence.flow.StateFlow;
-import org.hisp.dhis.client.sdk.android.api.persistence.flow.TrackedEntityDataValueFlow;
+import org.hisp.dhis.client.sdk.models.common.importsummary.ImportSummary;
 
 import java.util.List;
+import java.util.Map;
 
-/**
- * A static controller that orchestrate the push process
- * Created by arrizabalaga on 4/11/15.
- */
-public class PushController {
-    private final String TAG=".PushController";
-
-    private static PushController instance;
+public class PushController implements IPushController {
+    private final String TAG = ".PushController";
 
-    /**
-     * Context required to i18n error messages while pulling
-     */
-    private Context context;
+    private Context mContext;
+    private PushDhisSDKDataSource mPushDhisSDKDataSource;
+    private ConvertToSDKVisitor mConvertToSDKVisitor;
 
-    /**
-     * The stateful converter used to turn a survey into its corresponding event + datavalues;
-     */
-    public static ConvertToSDKVisitor converter;
 
-    /**
-     * Constructs and register this pull controller to the event bus
-     */
-    PushController(){
+    public PushController(Context context) {
+        mContext = context;
+        mPushDhisSDKDataSource = new PushDhisSDKDataSource();
+        mConvertToSDKVisitor = new ConvertToSDKVisitor(mContext);
     }
 
-    /**
-     * Singleton constructor
-     * @return
-     */
-    public static PushController getInstance(){
-        if(instance==null){
-            instance=new PushController();
-        }
-        return instance;
-    }
-    /**
-     * Flag that locks the push of events.
-     *
-     */
+    public void push(final IPushControllerCallback callback) {
 
-    public static boolean isPushing;
+        List<Survey> surveys = findSurveysToPush();
```

Comment:
This method is called when the user click on "refresh metadata" to force to delete/update the not sent surveys.
This query is not the background push query. 
You should change it by the correct query. ( it's in the next comment)


@xurxodev i think... you don't know about the two kind of push.

Background Push: it is the normal push, It pushes all the survey marked as completed. Query: Survey.getAllCompletedSurveys();
ProgressActivity push(it was partially removed in the last refactor):  when the user try to refresh the metadata, the app was sending the unsent surveys(completed or in porgress). Query: Survey.getAllUnsentUnplannedSurveys();